### PR TITLE
Expose pass-through of snapshotName argument for `matchSnapshot`

### PR DESCRIPTION
**Pull request by abmai**
This is poorly documented by Jest; but it is available. It allows users to provide custom names for snapshots so the following code

```js
test('it does render', (t) => {
  t.matchSnapshot('snapshot name');
});
```

Generates this snapshot name

```
exports[`it does render: snapshot name 1`] = ...
```

Documented here: https://facebook.github.io/jest/docs/en/expect.html#tomatchsnapshotpropertymatchers-snapshotname

And then the single-argument works because of here:
https://github.com/facebook/jest/blob/master/packages/jest-snapshot/src/index.js#L57

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -90,7 +90,9 @@ let mockFunction: MockFunctionType<*, *>, test: any;
 if (typeof it !== 'undefined') {
   // Surface snapshot testing
   // $FlowFixMe
-  assert.matchSnapshot = tree => expect(tree).toMatchSnapshot();
+  assert.matchSnapshot = (tree, snapshotName) =>
+    // $FlowFixMe
```
Comment (KevinGrandon) [inline src/index.js]: Why this $FlowFixMe?

Comment (abmai) [reply src/index.js]: Without this, `flow` was complaining that `expect` could not be found.

Comment (abmai) [reply src/index.js]: Any idea on how to better address that?

File: src/index.js [hunk 2]
```diff
@@ -78,7 +78,7 @@ type MockFunctionType<TArgs, TReturn> = () => $Call<
   ExtractArgsReturnType<TArgs, TReturn>,
   JestFnType
 >;
-type MatchSnapshotType = mixed => void;
+type MatchSnapshotType = (tree: mixed, snapshotName: ?string) => void;
```
Comment (abmai) [inline src/index.js]: FYI - added param name here for better clarifications.

File: src/index.js (hunk 1 above)
Comment (KevinGrandon) [reply src/index.js]: Ah, ok, I think this was maybe pre-existing then. Nice!

**Comment by KevinGrandon** [comment]
!merge

**Pull request merged**